contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,204 @@
+
+>Angular started as a side project. Back in 2009, Miško Hevery and Adam Abrons released a project that would help developers build web applications using simple HTML tags. The name “Angular” comes from the angle brackets, or < >, that surround all HTML tags.
+>
+>Miško described the idea behind the framework in an <a href="https://www.infoworld.com/article/2612801/javascript/whats-so-special-about-googles-angularjs.html">interview done in 2013</a>. 
+
+## Why learn Angular?
+
+Angular is a complex framework and can be challenging for new developers to understand.
+However, the advantages Angular provides justifies its steep learning curve.
+
+1. **Angular supports Single Page Applications (SPAs).**
+
+    <box>Single Page Applications are a type of web applications that load a single HTML page. The key feature is that SPAs can communicate with back-end servers without refreshing the full web page. SPAs load information on a single HTML page. The page updates dynamically through JavaScript as the user interacts with the app. This enhances the user experience.</box>
+ 
+2. **Angular uses TypeScript.** Angular applications are built using TypeScript language, a superset of JavaScript. TypeScript provides static typing. This helps the compiler show warnings about any potential errors in the code, before the app runs. Another advantage is that TypeScript provides code completion using [Intellisense](https://docs.microsoft.com/en-us/visualstudio/ide/using-intellisense?view=vs-2017). IntelliSense provides active hints as code is added.
+
+    <box> TypeScript was created and is maintained by Microsoft. It has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> over the last few years making it reliable to use.</box>
+                                                  
+3. **Google Long-Term Support.** Some software engineers consider the fact that Angular is supported by Google an advantage of the technology. While this may sound justified, it does not serve as much of an incentive. The good sign though is that Google announced Long-Term Support (LTS) for the technology back in 2017.
+
+    These are some of the reasons why you might want to get familiar with Angular.
+
+    [source](https://hackr.io/blog/why-should-you-learn-angular-in-2018)
+
+### Benefit 1: Angular provides a declarative user interface.
+
+Angular uses HTML to define the app’s user interface. HTML is a declarative language which is more intuitive than using imperative code to define the UI. An Imperative approach requires you to tell the compiler how you want the program to run by focussing on the logic implementation. Declarative code, on the other hand enables you to describe the logic without worrying about the implementation. How does it help? You do not need to invest time in dealing with how your program runs and deciding what to load first.
```

Comment:
The declarative vs imperative is something you can put inside a box, because it is an extra bit of background knowledge the user needs to understand the argument.